sdk/eventhubs/azure_messaging_eventhubs/src/consumer/event_receiver.rs

```diff
@@ -172,35 +119,24 @@ impl EventReceiver {
     ///
     pub fn stream_events(&self) -> impl Stream<Item = azure_core::Result<ReceivedEventData>> + '_ {
         // Use async_stream to create a stream that yields messages from the receiver.
-        try_stream! {
+        Box::pin(try_stream! {
             loop {
-                 let delivery = retry_azure_operation( || async move {
-                     if let Some(delivery_timeout) = self.timeout {
-                    select! {
-                        delivery = self.receiver.receive_delivery().fuse() => Ok(delivery),
-                        _ = azure_core::sleep::sleep(delivery_timeout).fuse() => {
-                             Err(azure_core::Error::new(
-                                AzureErrorKind::Io,
-                                Box::new(std::io::Error::from(std::io::ErrorKind::TimedOut))))
-                        },
-                    }?
-                 } else {
-                     self.receiver.receive_delivery().await
-                 }}, &self.retry_options, Some(Self::should_retry_receive_operation)).await?;
+                let receiver = self.connection.get_receiver(&self.source_url, self.message_source.clone(), self.receiver_options.clone(), self.timeout).await?;
```

Comment:
Surprise/nit: the formatter didn't wrap this long line? Just wondering more so if everything was formatted since some check might care.

Reply:
The rust formatter doesn't work consistently inside a macro (`try_stream!`).

No idea why but it doesn't - it's kinda annoying, tbh.

Fixing manually.

Reply:
> Fixing manually.

In the future, no need. I don't mind the formatting myself, but *when* it does apply automatically, keeping it consistent across clones is nice. But, yeah, if it didn't format here and that's consistent across machines (which I expect), I actually didn't mind the long line. It was just a red (or, rather, yellow) flag - a possible symptom of something else wrong.

Reply:
100% agreed. But it's annoying to me, especially on my home machine (vertical monitor) so I manually updated it.